### PR TITLE
Balder/move commit time tracker to replay state

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge PR

### CODE REVIEW

**Review by toregge — APPROVED** [review]
Consider changing CommitTimeTracker::needCommit() to return false if visibility delay is zero. The feed operation should already have been committed in that case due to StoreOnlyFeedView::needCommit() having returned true.

File: searchcore/src/tests/proton/documentdb/configurer/configurer_test.cpp [hunk 1]
```diff
@@ -267,16 +262,16 @@ struct MyFastAccessFeedView
...
+        _lidReuseDelayer = std::make_unique<documentmetastore::LidReuseDelayer>(_writeService, _dmsc->get());
         std::shared_ptr<const DocumentTypeRepo> repo = createRepo();
         StoreOnlyFeedView::Context storeOnlyCtx(summaryAdapter, schema, _dmsc, *_gidToLidChangeHandler, repo,
-                                                _writeService, *_lidReuseDelayer, _commitTimeTracker);
+                                                _writeService, *_lidReuseDelayer);
         StoreOnlyFeedView::PersistentParams params(1, 1, DocTypeName(DOC_TYPE), 0, SubDbType::NOTREADY);
         auto mgr = make_shared<AttributeManager>(BASE_DIR, "test.subdb", TuneFileAttributes(), _fileHeaderContext,
                                                  _writeService.attributeFieldWriter(), _writeService.shared(), _hwInfo);
-        IAttributeWriter::SP writer(new AttributeWriter(mgr));
+        auto writer = std::make_shared<AttributeWriter>(mgr);
```
Comment (toregge) [inline searchcore/src/tests/proton/documentdb/configurer/configurer_test.cpp:272]: Use of auto here triggers a memory use after free error (detected by address sanitizer).

**Review by toregge — CHANGES_REQUESTED** [review]
Please fix the stack use after scope error.

File: searchcore/src/tests/proton/documentdb/configurer/configurer_test.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/tests/proton/documentdb/configurer/configurer_test.cpp:272]: How does that change scope ?

File: searchcore/src/vespa/searchcore/proton/server/storeonlyfeedview.h [hunk 2]
```diff
@@ -81,24 +81,21 @@ class StoreOnlyFeedView : public IFeedView,
         const std::shared_ptr<const document::DocumentTypeRepo>    &_repo;
         searchcorespi::index::IThreadingService &_writeService;
         documentmetastore::ILidReuseDelayer     &_lidReuseDelayer;
-        CommitTimeTracker                       &_commitTimeTracker;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/storeonlyfeedview.h:84]: Fixed.

File: searchcore/src/tests/proton/documentdb/configurer/configurer_test.cpp (hunk 1 above)
Comment (toregge) [reply searchcore/src/tests/proton/documentdb/configurer/configurer_test.cpp:272]: A reference to a temporary std::shared_pointer<IattributeWriter> is passed to FastAccessFeedView::Context constructor and goes out of scope right afterwards, causing fastUpdateCtx to reference a dead stack variable.

**Review by toregge — APPROVED** [review]

File: searchcore/src/tests/proton/documentdb/configurer/configurer_test.cpp (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/tests/proton/documentdb/configurer/configurer_test.cpp:272]: Fixed implementation of ...::Context to use shared_ptr by value instead.

**Comment by baldersheim** [comment]
@toregge Now it should be better. PTAL

**Review by toregge — APPROVED** [review]

**Review by toregge — APPROVED** [review]

**Pull request merged**